.github/workflows/main.yml

```diff
@@ -0,0 +1,83 @@
+name: CI
+
+on: pull_request
+
+jobs:
+  jjb:
+    runs-on: ubuntu-latest
+    name: Jenkins Job Builder
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '2.7'
+      - name: Restore pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements*.txt') }}
```

Comment:
Can you include the Python version in the cache key? That way the cache is automatically invalidated when we switch the Python version.

Reply:
The cached wheels are version agnostic, so why bother?

Reply:
I suspected lxml was version dependent. It's not?

Reply:
lxml is, indeed, but then the filename in the cache is different too and it'll be updated from the internet. no idea what's saving more, to be honest.

Reply:
I suspect that if you use a common cache, it can grow and grow. Not sure if there's anything that purges old wheels from the cache, but switching Python version could be a good enough thing to force that.